### ISSUE TITLE
Request trace data is continually getting appended 

### ISSUE THREAD

**Issue by donaldlittlepie** [glowroot/glowroot#1]
Looks like the trace data is continually getting appended, so data from one request appears in the trace for another request.  Started seeing this error in the logs with glowroot running on the latest version.  This wasn't happening in previous builds.

```
java.lang.RuntimeException: java.lang.ClassNotFoundException: org.apache.karaf.jaas.modules.RolePrincipal not found by org.apache.cxf.cxf-rt-transports-http [141]
        at org.eclipse.jetty.plus.jaas.JAASLoginService.getGroups(JAASLoginService.java:306)[118:org.eclipse.jetty.plus:7.5.4.v20111024]
        at org.eclipse.jetty.plus.jaas.JAASLoginService.login(JAASLoginService.java:223)[118:org.eclipse.jetty.plus:7.5.4.v20111024]
        at org.eclipse.jetty.security.authentication.BasicAuthenticator.validateRequest(BasicAuthenticator.java:83)[113:org.eclipse.jetty.security:7.5.4.v20111024]
        at org.eclipse.jetty.security.authentication.DeferredAuthentication.authenticate(DeferredAuthentication.java:103)[113:org.eclipse.jetty.security:7.5.4.v20111024]
        at org.eclipse.jetty.server.Request.getUserPrincipal(Request.java:1226)[116:org.eclipse.jetty.server:7.5.4.v20111024]
        at org.glowroot.plugin.servlet.ServletAspect$ServiceAdvice.onBefore(ServletAspect.java:142)[:0.8-SNAPSHOT]
        at org.apache.cxf.transport.servlet.AbstractHTTPServlet.service(AbstractHTTPServlet.java)[141:org.apache.cxf.cxf-rt-transports-http:2.6.8]
```

**Comment by trask** [comment]
Thanks for the report!

I was able to repro and added test coverage and fix 442efcedac761d43da3cf0b19d9e7095ba1344fe

Please try latest snapshot when you have a chance https://glowroot.s3.amazonaws.com/snapshots/latest/glowroot-dist.zip

**Comment by trask** [comment]
Released in 0.8.  Thanks!
